api/envoy/extensions/filters/http/grpc_field_extraction/v3/config.proto

```diff
@@ -0,0 +1,131 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.grpc_field_extraction.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.grpc_field_extraction.v3";
+option java_outer_classname = "ConfigProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/grpc_field_extraction/v3;grpc_field_extractionv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: gRPC Field Extraction]
+// gRPC Field Extraction :ref:`configuration overview
+// <config_http_filters_grpc_field_extraction>`.
+// [#extension: envoy.filters.http.grpc_field_extraction]
+
+// GrpcFieldExtractionConfig supports extracting the fields from first gRPC
+// request message no matter if it is unary or streaming and inject the result
+// as a header. It won't propagate headers until the extraction and the
+// injection done so the following filters can use it. No op if isn't a gRPC
+// method or its type cannot be found in the proto descriptor.
+//
+// The target field should be singular primitive type or repeated primitive
+// type and its value will be extracted in string format.
+//
+//
+// The output is a serialized google.protobuf.Struct of field<StringValue> -> extract values<ListValue of StringValue>. It will be injected in the `header` with base64 encoded format.
```

Comment:
I think it might be better to rely on https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/header_to_metadata/v3/header_to_metadata.proto#envoy-v3-api-enum-extensions-filters-http-header-to-metadata-v3-config-valueencode to encode the value as a header. An internal metadata format is more efficient than encoding/decoding to headers.

Reply:
Filter metadata definitely is in the scope for this filter to support in the next future and ACK it is more efficient. We will use an extAuthz filter using the header added by this filter so metadata won't work in this callout case.

Reply:
extAuthz supports passing dynamic metadata AFAIK. unless the backend needs the header (which is unlikely since it's in the body), the recommended use case should be using dynamic metadata.

Reply:
Thanks! Changed to dynamic metadata now.